### PR TITLE
refactor: improved modal

### PR DESCRIPTION
**Pull request by vatsalsinghkv**
## Fixes Issue 

Closes: #612 

#### Describe the changes you've made ✍
- Created new component for modal as `Modal.jsx`
- Wrote css for modal component in `Modal.module.css`
- Created a portal which will in insert modal at `#modal` next to `#root` (not nested)
- Converted modal's border radius to `px` from `%` so we can have the same design independent of the modal's height and width

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -54,6 +54,7 @@
 <body>
   <noscript>You need to enable JavaScript to run this app.</noscript>
   <div id="root"></div>
+  <div id="modal"></div>
```
Comment (IAmTamal) [inline index.html:57]: I just wanna ask, what's the reason for adding the modal here ?

**Review by IAmTamal — COMMENTED** [review]
The rest looks good to me !

File: index.html (hunk 1 above)
Comment (vatsalsinghkv) [reply index.html:57]: Actually I am creating a portal, means modal will get rendered into this `#modal` doesn't matter which nested component implement the modal. This is an efficient way to create modals, for example our modal has `z-index: 20` if we the component who's nested like more than 20 level deep then our modal will not be on the top. So to solve this we use portal to render modal on the root level element

Read more on this:
https://react.dev/reference/react-dom/createPortal#rendering-a-modal-dialog-with-a-portal

Comment (IAmTamal) [reply index.html:57]: Oh , now i get it ! 
Thanks for letting me know about this.

**Amazing work,**
This looks good to me, will be merged soon 🎉.

**Review by IAmTamal — APPROVED** [review]
**Amazing work,**
This looks good to me, will be merged soon 🎉.

**Pull request merged**